File: Anul II/Medii de proiectare si programare/Laborator/lab06/services/src/main/java/com/ubb/IContestServices.java
package com.ubb;

import com.ubb.exceptions.ContestDataException;
import com.ubb.model.*;
import com.ubb.model.data.RaceDTO;

import java.util.Collection;
import java.util.List;
import java.util.Optional;

public interface IContestServices {
    void login(String username, String passwordToken, IMainObserver client) throws ContestDataException;

    void logout(User user, IMainObserver client) throws ContestDataException;

    Collection<Participant> getParticipantsByTeam(final Long teamID) throws ContestDataException;

    Optional<Participant> getParticipantByData(final Participant participant) throws ContestDataException;

    Optional<Race> getRaceByName(String name) throws ContestDataException;

    Collection<RaceEntry> getEntriesByRace(Long raceID) throws ContestDataException;

    Collection<Race> getRacesByEngineCapacity(Integer engineCapacity) throws ContestDataException;

    Collection<RaceDTO> getRacesWithParticipantCount() throws ContestDataException;

    void saveRaceEntries(List<RaceEntry> newEntities) throws ContestDataException;

    Collection<Race> getRacesWhereNotRegisteredAndEngineCapacity(Long participantID, Integer engineCapacity) throws ContestDataException;

    Optional<Team> getTeamByName(String teamName) throws ContestDataException;

    void saveParticipant(final Participant newEntity) throws ContestDataException;

    void saveRace(final Race newEntity) throws ContestDataException;
}

File: Anul II/Medii de proiectare si programare/Laborator/lab08/C#/Ubb.BikeContest.Service/IContestServices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Ubb.BikeContest.Model;
using Ubb.BikeContest.Model.data;

namespace Ubb.BikeContest.Services
{
    public interface IContestServices
    {
        User Login(string username, string passwordToken, IMainObserver client);
        void Logout(User user, IMainObserver client);
        List<RaceDto> GetRacesWithParticipantCount();
        List<Team> FindAllTeams();
        List<Participant> GetParticipantsByTeam(long id);
        Participant GetParticipantByData(Participant participant);
        List<Participant> FindAllParticipants();
        void SaveRaceEntries(List<RaceEntry> newEntities);
        void SaveParticipant(Participant newEntity);
        List<Race> GetRacesWhereNotRegisteredAndEngineCapacity(long participantId, int engineCapacity);
        List<RaceEntry> GetEntriesByRace(long raceId);
        Race GetRaceByName(string name);
        List<Race> GetRacesByEngineCapacity(int engineCapacity);
        Team GetTeamByName(string name);
        void SaveRace(Race race);
    }
}
